coordinator/ethereum/gas-pricing/static-cap/src/main/kotlin/net/consensys/linea/ethereum/gaspricing/staticcap/HistoricVariableCostProviderImpl.kt

```diff
@@ -0,0 +1,70 @@
+package net.consensys.linea.ethereum.gaspricing.staticcap
+
+import linea.OneKWei
+import linea.domain.BlockParameter
+import linea.domain.BlockParameter.Companion.toBlockParameter
+import linea.kotlin.encodeHex
+import linea.web3j.ExtendedWeb3J
+import net.consensys.linea.ethereum.gaspricing.HistoricVariableCostProvider
+import net.consensys.linea.ethereum.gaspricing.MinerExtraDataV1
+import org.apache.logging.log4j.LogManager
+import org.apache.logging.log4j.Logger
+import tech.pegasys.teku.infrastructure.async.SafeFuture
+import java.util.concurrent.atomic.AtomicReference
+
+class HistoricVariableCostProviderImpl(
+  private val web3jClient: ExtendedWeb3J,
+) : HistoricVariableCostProvider {
+  private val log: Logger = LogManager.getLogger(this::class.java)
+  private var lastVariableCost: AtomicReference<Pair<ULong, Double>> =
+    AtomicReference(0UL to 0.0)
+
+  private fun getHistoricVariableCostInWei(blockParameter: BlockParameter): SafeFuture<Double> {
+    return web3jClient.ethGetBlock(blockParameter)
+      .thenApply { block ->
+        try {
+          MinerExtraDataV1.decodeV1(block!!.extraData.encodeHex())
+            .variableCostInKWei.toDouble() * OneKWei
+        } catch (th: Throwable) {
+          if (block != null) {
+            log.debug(
```

Comment:
Should it be a WARN?

Reply:
I'd keep it as debug as it does happen during the first few blocks in our local stack (before we set the first miner extra data for variable cost)